hazelcast/src/main/java/com/hazelcast/map/impl/NearCache.java

```diff
@@ -63,6 +66,9 @@
     private final NearCacheStatsImpl nearCacheStats;
     private final SerializationService serializationService;
     private final Comparator<NearCacheRecord> selectedComparator;
+    private final Object announcementLock = new Object();
+    private volatile long announcementPermitSeq;
```

Comment:
`announcementPermitSeq` is only used (increased and get) in a `synchronized` block in `announcePut` method. So do we really need to define it as `volatile`?